codegen/mock_can_data.py

```diff
@@ -0,0 +1,77 @@
+
+""" Script to mock CAN data """
+# This script automatically generates and sends CAN messages based off of the DBC file
+# Ensure that you have can and cantools installed, which can be accomplished by running
+# pip install -r requirements.txt
+# Alternatively, you could also just run
+# pip install can && pip install cantools
+
+# If you are running this script with virtual CAN ensure that it is set up first
+# You can run the below commands
+# sudo modprobe vcan
+# sudo ip link add dev vcan0 type vcan
+# sudo ip link set up vcan0
+
+import time
+import random
+import cantools
+import can
+
+# A negative value for NUM_MESSAGES will cause the script to send CAN messages forever
+NUM_MESSAGES = -1
+SLEEP_TIME_S = 1
+CAN_MESSAGES = []
+
+try:
+    DB = cantools.database.load_file('system_can.dbc')
+# pylint: disable=broad-except
+except BaseException:
+    print("Must generate DBC file first")
+    print("Run make codegen && make codegen_dbc")

```

Comment:
IMO you should `raise` or `sys.exit(1)` at the end of this `except` block - otherwise you get confusing errors later on when the script tries to use `DB`.